pegasus/sites.v3/code.org/public/congrats.haml

```diff
@@ -49,10 +49,15 @@ max_age: 60
   -else
     %h1=I18n.t :congratulations_on_completing_hoc
 
-  -if request.locale == 'ar-SA'
-    %h2
-      أنهيت ساعة البرمجة ! العودة إلى
-      %a{:href=>"http://click-and-share.com/hour_of_code/data-capture/"} كرتون نتورك
+  -if params[:partner] == 'CN' && cert_script == 'gumball'
```

Comment:
You can rewrite this as a client-side code that looks at the cookie and decides to show or hide a div with the cartoon network link.